src/vellum/workflows/runner/runner.py

```diff
@@ -117,6 +120,8 @@ def __init__(
         self._is_resuming = False
         self._should_emit_initial_state = True
         self._span_link_info: Optional[Tuple[str, str, str, str]] = None
+        # Track if caller explicitly provided entrypoints
+        self._caller_provided_entrypoints = bool(entrypoint_nodes or external_inputs)
```

Comment:
[nit] I think we can avoid tracking this flag if we just add an `elif trigger` case in this if/else tree below